### ISSUE TITLE
Invalid linter name

### ISSUE THREAD

**Issue by renkun-ken** [jimhester/lintr#786]
With the latest `master`, Looks like some linters do not correctly specify their linter name in the result. For example, `infix_spaces_linter`:

```r
> str(lintr::lint("x<-1\n"))                                                                                                         
List of 1
 $ <text>:List of 8
  ..$ filename     : chr "<text>"
  ..$ line_number  : int 1
  ..$ column_number: int 2
  ..$ type         : chr "style"
  ..$ message      : chr "."
  ..$ line         : Named chr "x<-1"
  .. ..- attr(*, "names")= chr "1"
  ..$ ranges       :List of 1
  .. ..$ : int [1:2] 1 4
  ..$ linter       : chr "function ()  {     Linter(function(source_file) {         lapply(ids_with_token(source_file, infix_tokens, fun "| __truncated__
  ..- attr(*, "class")= chr "lint"
 - attr(*, "class")= chr "lints"
 - ```

**Comment by renkun-ken** [comment]
It turns out to be caused by my `~/.lintr` contains a `infix_spaces_linter` instead of `infix_spaces_linter()`.
